src/pages/workspace/accounting/qbo/import/QuickbooksLocationsPage.tsx

```diff
@@ -20,7 +20,8 @@ function QuickbooksLocationsPage({policy}: WithPolicyProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
     const policyID = policy?.id ?? '';
-    const {syncLocations, pendingFields} = policy?.connections?.quickbooksOnline?.config ?? {};
+    const {syncLocations, pendingFields, reimbursableExpensesExportDestination} = policy?.connections?.quickbooksOnline?.config ?? {};
+    const shouldBeDisabled = !syncLocations && reimbursableExpensesExportDestination !== CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.JOURNAL_ENTRY;
```

Comment:
Looks like we need to use `isSwitchOn` instead of `syncLocations`. Else the `locations` toggle would not get disabled. 
`    const shouldBeDisabled = !isSwitchOn && reimbursableExpensesExportDestination !== CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.JOURNAL_ENTRY;`

Reply:
ah you're right! because we have this logic `syncLocations !== CONST.INTEGRATION_ENTITY_MAP_TYPES.NONE` which `syncLocations` already incorporates 👍

Reply:
ah like this?

<img width="400" alt="Screenshot 2024-05-23 at 3 45 14 PM" src="https://github.com/Expensify/App/assets/98560306/152f6a2e-2d51-480c-b296-736a564ad0fc">